common/script_values/babylon_values.txt

```diff
@@ -1574,3 +1679,41 @@ babylon_weekly_customs_union_minting_cost = {
 		subtract = modifier:country_minting_add
 	}
 }
+
+babylon_china_infamy_penalty_value = {
+	value = 0
+	if = {
+		limit = {
+			root = {
+				NOR = {
+					country_has_primary_culture = cu:manchu
+					country_has_primary_culture = cu:mongol
+					country_has_primary_culture = cu:tibetan
+					any_primary_culture = {
+						has_discrimination_trait = han_chinese
+					}
+				}
+			}
+		}
+		every_state = {
+			limit = {
+				OR = {
+					owner = root
+					owner = {
+						any_overlord_or_above = {
+							this = root
+						}
+					}
+				}
+				is_treaty_port = no
+				state_region = {
+					is_homeland = cu:han
+					NOT = {
+						this = s:STATE_FORMOSA
+					}
+				}
+			}
+			add = 5
+		}
+	}
```

Comment:
this is way to inefficient. I mean, this is a monthly states * country check.

Reply:
Ideally you just check every state in china, once.

Reply:
I have tried for 2 hours now to get it to work using an every_state call in a single country. There _seems_ to be a bug where in on_actions, if you go from country _a_ to country _b_ in the on_actions, and use a variable as a multiplier in a modifier, it'll take that variable from country _a_ and not country _b_.
It should be said that the same code worked when used in the ingame script editor, in the country screen.

I have the script I used saved, in case you want to take a look at it, but.. This solution, while not optimal, at least works.